fix: improve pixel format selection for video output on iPhone 17

nngb102 opened the pull request:
# Improve Pixel Format Selection for Video Output on iPhone 17

## Problem

iPhone 17 devices (iOS 26.0.1) crash when starting MobileScanner with the error:
```
-[AVCaptureVideoDataOutput setVideoSettings:] Unsupported pixel format type - use -availableVideoCVPixelFormatTypes
```

**Root Cause**: The code was hardcoding a pixel format (`kCVPixelFormatType_32BGRA`) without checking which formats are actually supported by the device. On iPhone 17, this format is not available, causing the crash.

## Solution

Updated the pixel format selection logic in `start()` to:

1. **Check available formats**: Use `availableVideoPixelFormatTypes` from `AVCaptureVideoDataOutput` to query supported pixel formats on the device.

2. **Prioritize formats in order**:
   - `kCVPixelFormatType_32BGRA`
   - `kCVPixelFormatType_420YpCbCr8BiPlanarFullRange`
   - `kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange`

3. **Fallback**: If none of the preferred formats are available, select the first available format from the device's supported formats.

This ensures we only set a pixel format that the device supports, preventing crashes on iPhone 17 and newer devices while maintaining backward compatibility with older iOS devices.

## Code Changes

- **Location**: `darwin/mobile_scanner/Sources/mobile_scanner/MobileScannerPlugin.swift` (lines 461-495)
- **Change**: Replaced hardcoded pixel format assignment with dynamic format selection logic that checks device capabilities before setting `videoSettings`.

## Result

- ✅ No more crashes on iPhone 17 when starting the scanner
- ✅ Backward compatible with older iOS devices
- ✅ Automatically selects the best available format based on device capabilities

## References

- Issue: https://github.com/juliansteenbakker/mobile_scanner/issues/1578
- Similar issue reported in Flutter camera package

Fixes #1578

nngb102 committed:
fix: improve pixel format selection for video output on iPhone 17

navaronbracke commented on darwin/mobile_scanner/Sources/mobile_scanner/MobileScannerPlugin.swift:
```diff
@@ -457,7 +457,43 @@ public class MobileScannerPlugin: NSObject, FlutterPlugin, FlutterStreamHandler,
-        videoOutput.videoSettings = [kCVPixelBufferPixelFormatTypeKey as String: kCVPixelFormatType_32BGRA]
+        
+        // Fix for iPhone 17: Check available pixel formats before setting
+        // Issue: https://github.com/juliansteenbakker/mobile_scanner/issues/1578
```
We do not specifically need a link to the issue (links can get stale). I would omit the link here.

navaronbracke commented on darwin/mobile_scanner/Sources/mobile_scanner/MobileScannerPlugin.swift:
```diff
@@ -457,7 +457,43 @@ public class MobileScannerPlugin: NSObject, FlutterPlugin, FlutterStreamHandler,
+        }
+        
+        // Apply the selected format or fallback to the old default
+        if let format = selectedFormat {
```
Can we put the format selection into a separate method, so that we can do

```
let format = getPreferredVideoFormat(videoOutput: videoOutput)

videoOutput.videoSettings = [kCVPixelBufferPixelFormatTypeKey as String: format]
```

navaronbracke reviewed (changes requested):
Thanks for this fix! This will also need a patch version bump and a changelog entry, so that we can publish the fix.

navaronbracke commented:
Edited the OP, so that the linked issue will be closed once this is merged.

nngb102 committed:
fix: improve pixel format selection for video output on iPhone 17

nngb102 commented:
@navaronbracke 
Thanks for the review! I've addressed all the feedback:

- Removed the issue link from the code comment
- Extracted the format selection logic into a separate `getPreferredVideoFormat` method
- Bumped version to 7.1.4 and added changelog entry

Ready for review!

navaronbracke commented:
Version 7.1.4 should be available on Pub.dev now.
